### PR TITLE
Implement kubernetes auth / Add customizable auth path

### PR DESCRIPTION
**Pull request by FalconerTC**
Built from https://github.com/hashicorp/vault-action/pull/155. Thanks to jojo19893  for most of the work.

This adds support for Kubernetes based auth (https://www.vaultproject.io/docs/auth/kubernetes).  It also adds a parameter to customize the path for any auth method.

### CODE REVIEW

**Comment by FalconerTC** [comment]
@jasonodonnell Any chance for a review on this sometime soon?

File: action.yml [hunk 1]
```diff
@@ -14,6 +14,12 @@ inputs:
     description: 'The method to use to authenticate with Vault.'
     default: 'token'
     required: false
+  role:
+    description: 'Vault role for specified auth method'
+    required: false
+  path:
+    description: 'Custom vault path, if the auth method was enabled at a different path'
```
Comment (jasonodonnell) [inline action.yml:21]:
```suggestion
    description: 'Custom Vault path, if the auth method was mounted at a different path'
```

File: action.yml [hunk 2]
```diff
@@ -26,6 +32,10 @@ inputs:
   githubToken:
     description: 'The Github Token to be used to authenticate with Vault'
     required: false
+  kubernetesTokenPath:
+    description: 'The path to the kubernetes service account secret'
```
Comment (jasonodonnell) [inline action.yml:36]:
```suggestion
    description: 'The path to the Kubernetes service account secret'
```

**Comment by jasonodonnell** [comment]
Hi @FalconerTC, thanks for the contribution! This looks great at a glance, but I need to test this before merging. I'm guessing the idea here is to use a remote runner on K8s?

**Comment by FalconerTC** [comment]
Yep. I use the following tool (https://github.com/actions-runner-controller/actions-runner-controller) currently for self-hosted runners on K8s.

**Comment by FalconerTC** [comment]
@jasonodonnell Any chance you will be able to look at this again this soon? I've validated it authenticates correctly in a private workflow, including validating with a custom path

**Review by jasonodonnell — APPROVED** [review]
LGTM, thanks!

**Pull request merged**